packages/agent_cli/src/main.rs

```diff
@@ -1,618 +1,699 @@
-                )
-                .subcommand(
-                    Command::new("list")
-                        .about("List tunnels (format \"[tunnel-id] [port-type] [port-count] [public-address]\")")
-                )
-        )
-        .subcommand(
-            Command::new("run")
-                .about("Run the playit agent")
-                .arg(arg!([MAPPING_OVERRIDE] "(format \"<tunnel-id>=[<local-ip>:]<local-port> [, ..]\")").required(false).value_delimiter(','))
-        )
-        .subcommand(
-            Command::new("reset")
-                .about("removes the secret key on your system so the playit agent can be re-claimed")
-        )
-        .subcommand(
-            Command::new("secret-path")
-                .about("shows the file path where the playit secret can be found")
-        )
-        ;
-
-    #[cfg(target_os = "linux")] {
-        cmd = cmd.subcommand(Command::new("setup"));
-    }
-
-    cmd
-}
+use std::collections::{HashMap, HashSet};
+use std::error::Error;
+use std::fmt::{Display, Formatter};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4};
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::Duration;
+
+use clap::{arg, Command};
+use rand::Rng;
+use uuid::Uuid;
+
+use autorun::autorun;
+use playit_agent_core::api::api::*;
+use playit_agent_core::api::http_client::HttpClientError;
+use playit_agent_core::api::PlayitApi;
+use playit_agent_core::network::address_lookup::{AddressLookup, AddressValue};
+use playit_agent_core::tunnel::setup::SetupError;
+use playit_agent_core::tunnel_runner::TunnelRunner;
+use playit_agent_core::utils::now_milli;
+use playit_secret::PlayitSecret;
+
+use crate::match_ip::MatchIp;
+use crate::signal_handle::get_signal_handle;
+use crate::ui::{UI, UISettings};
+
+
+//imports (chroma)
+use std::process::Command as ProcessCommand;
+use std::str;
+use crossterm::style::Stylize;
+
+pub const API_BASE: &'static str = "https://api.playit.gg";
+
+pub mod util;
+pub mod autorun;
+pub mod playit_secret;
+pub mod match_ip;
+pub mod ui;
+pub mod signal_handle;
+
+
+
+
+
+//ping function !WINDOWS! (chroma)
+async fn ipcheck() -> Result<(), CliError> {
+
+    println!("Testing {}","ply.gg...".yellow());
+    let output = ProcessCommand::new("ping")
+        .arg("-n")
+        .arg("1")
+        .arg("ping.ply.gg")
+        .output();
+
+    match output {
+        Ok(output) if output.status.success() => {
+            println!("PLY.gg is {}","not blocked!".green());
+            check_ips("ip", &["na", "sa", "eu", "as", "in"]).await;
+        }
+        _ =>{
+            println!("PLY.gg might be {}","blocked!".red());
+            check_ips("209.25.140", &[""]).await;
+            check_ips("209.25.141", &[""]).await;
+            check_ips("209.25.142", &[""]).await;
+            check_ips("209.25.143", &[""]).await;
+            check_ips("23.133.216", &[""]).await;
+        }
+    }
+
+    Ok(())
+}
+
+async fn check_ips(base: &str, domains: &[&str]) {

```

Comment:
I think it would be cleaner to have this function return a `Result<bool, std::io::Error>` where the bool represents if the ping was successful and an error is thrown if you cannot determine the results.

The caller of the function would decide how to consider the response. So instead of printing debug statements here the caller would do that.